miqa/learning/nn_classifier.py

```diff
@@ -302,6 +305,45 @@ def evaluate1(model, image_path):
     return labeled_results
 
 
+def evaluate_many(model, image_paths):
+    itk_reader = monai.data.ITKReader()
+    # Define transforms for image
+    train_transforms = Compose(
+        [
+            LoadImaged(keys=['img'], reader=itk_reader),
+            EnsureChannelFirstd(keys=['img']),
+            ScaleIntensityd(keys=['img']),
+            ToTensord(keys=['img']),
+        ]
+    )
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+    evaluation_files = [
+        {
+            'img': image_path,
+            'info': torch.FloatTensor([0] * (regression_count + len(artifacts))),
+        }
+        for image_path in image_paths
+    ]
+
+    evaluation_ds = monai.data.Dataset(evaluation_files, transform=train_transforms)
+    evaluation_loader = DataLoader(evaluation_ds, pin_memory=torch.cuda.is_available())
+    results = evaluate_model(model, evaluation_loader, device, None, 0, 'evaluate_many')
+
+    labeled_results = {}
+    for index, result in enumerate(results):
+        corresponding_image_path = image_paths[index]
+        single_labeled = {
+            'overall_quality': result[0],
+            'signal_noise_ratio': result[1],
+            'contrast_noise_ratio': result[2],
+        }
+        for index, artifact_name in enumerate(artifacts):
+            single_labeled[artifact_name] = result[index + 3]
```

Comment:
Alternatively, you could do:

```suggestion
        for artifact_name, value in zip(artifacts, result[regression_count:]):
            single_labeled[artifact_name] = value
```